### PR TITLE
automating setup of rosa hcp prerequisites

### PR DESCRIPTION
**Pull request by xinredhat**

### CODE REVIEW

File: ci/hack/rosa_hcp_setup.sh [hunk 1]
```diff
@@ -4,20 +4,23 @@
 set -o errexit
 set -o nounset
 set -o pipefail
+set -x
```
Comment (Roming22) [inline ci/hack/rosa_hcp_setup.sh:7]: This should be removed, that's what the `--debug` option is there for.

File: ci/hack/rosa_hcp_setup.sh [hunk 2]
```diff
@@ -75,69 +72,73 @@ parse_args() {
...
 }
 
 main() {
     parse_args "$@"
     prechecks
-    init
     create_vpc
-    create_account_roles
+    create_resources
 }
```
Comment (Roming22) [inline ci/hack/rosa_hcp_setup.sh:142]: You should clean the TMPDIR at the end.

File: ci/hack/rosa_hcp_setup.sh [hunk 3]
```diff
@@ -4,20 +4,23 @@
...
-  cd "$SCRIPT_DIR/../.." >/dev/null || exit 1
-  pwd
+    cd "$SCRIPT_DIR/../.." >/dev/null || exit 1
+    pwd
 )"
 
 # shellcheck source=ci/images/ci-runner/hack/bin/utils.sh
 source "$PROJECT_DIR/ci/images/ci-runner/hack/bin/utils.sh"
 
+TMPDIR=$(dirname "$(mktemp -u)")
```
Comment (Roming22) [inline ci/hack/rosa_hcp_setup.sh:22]: That should probably be in init()

File: ci/hack/rosa_hcp_setup.sh [hunk 4]
```diff
@@ -75,69 +72,73 @@ parse_args() {
...
+        usage
+        exit 1
     fi
-    if [[ -z "${AWS_REGION:-}" ]]; then
-      printf "[ERROR] AWS region is not set\n\n" >&2
-      usage
-      exit 1
+
+    if [[ -z "${AWS_ACCESS_KEY_ID:-}" ]]; then
+        printf "[ERROR] AWS_ACCESS_KEY_ID variable is not exported\n\n" >&2
```
Comment (Roming22) [inline ci/hack/rosa_hcp_setup.sh:84]: You use `is not set` or `is not exported` when vars are not set. Keep the wording consistent or users will be confused.

File: ci/hack/rosa_hcp_setup.sh [hunk 5]
```diff
@@ -75,69 +72,73 @@ parse_args() {
...
+    git clone https://github.com/openshift-cs/terraform-vpc-example.git
+    cd terraform-vpc-example
 
     # Initialize Terraform
     terraform init
 
     # Plan the Terraform deployment and save the plan to a file
-    terraform plan -out rosa.plan -var aws_region="$AWS_REGION" -var cluster_name="${PREFIX_NAME}"
+    terraform plan -out rosa.tfplan -var region="$AWS_REGION" -var cluster_name="${CLUSTER_NAME}"
+    # terraform plan -out rosa.tfplan -var aws_region="$AWS_REGION" -var cluster_name="${CLUSTER_NAME}"
```
Comment (Roming22) [inline ci/hack/rosa_hcp_setup.sh:117]: Remove that old command

Comment (xinredhat) [reply ci/hack/rosa_hcp_setup.sh:117]: fixed

File: ci/hack/rosa_hcp_setup.sh (hunk 1 above)
Comment (xinredhat) [reply ci/hack/rosa_hcp_setup.sh:7]: fixed

File: ci/hack/rosa_hcp_setup.sh (hunk 3 above)
Comment (xinredhat) [reply ci/hack/rosa_hcp_setup.sh]: fixed

File: ci/hack/rosa_hcp_setup.sh (hunk 4 above)
Comment (xinredhat) [reply ci/hack/rosa_hcp_setup.sh]: fixed

File: ci/hack/rosa_hcp_setup.sh [hunk 6]
```diff
@@ -28,36 +18,44 @@ Configure resources for ROSA HCP cluster on AWS
...
+        Name for the cluster name.
     
 Optional arguments:
     -d, --debug
         Activate tracing/debug mode.
     -h, --help
         Display this message.
 Example:
-    ${0##*/} ./rosa_hcp_setup.sh --prefix test0612 -r us-east-1 -v 4.12
+    ${0##*/} ./rosa_hcp_setup.sh --name test0612 -r us-east-1
```
Comment (Roming22) [inline ci/hack/rosa_hcp_setup.sh:30]: What's the mean of `0612`? Is it just a random string?

**Review by Roming22 — COMMENTED** [review]
Small comment, but otherwise lgtm.

**Comment by Roming22** [comment]
@xinredhat There's one small comment that needs your attention, and I suppose the version now needs to be bumped to `4.14`.

**Comment by xinredhat** [comment]
> @xinredhat There's one small comment that needs your attention, and I suppose the version now needs to be bumped to `4.14`.

@Roming22 For the setup of rosa hcp prerequisites, it doesn't care about HCP version. We just need to specify the HCP version while creating HCP cluster.

File: ci/hack/rosa_hcp_setup.sh (hunk 6 above)
Comment (xinredhat) [reply ci/hack/rosa_hcp_setup.sh]: fixed.

**Review by Roming22 — APPROVED** [review]

**Pull request merged**